dpnp/dpnp_algo/dpnp_algo_indexing.pyx

```diff
@@ -67,7 +67,9 @@ ctypedef void(*fptr_dpnp_nonzero_t)(const void * , void * , const size_t, const
 
 
 cpdef utils.dpnp_descriptor dpnp_choose(utils.dpnp_descriptor input, list choices1):
-    cdef shape_type_c input_shape = input.shape  
+    cdef shape_type_c input_shape = input.shape 
+    cdef shape_type_c choices_shape = (len(choices1),) 
+    cdef shape_type_c choice_shape = choices1[0].shape
```

Comment:
Looks like we don't need the `shapes`, because we use only first element from them.